move/axelar_gateway/sources/gateway.move

```diff
@@ -212,27 +176,14 @@ entry fun rotate_signers(
     proof_data: vector<u8>,
     ctx: &TxContext,
 ) {
-    self.version_control.check(VERSION, b"rotate_signers");
-    let weighted_signers = utils::peel!(
+    let data = self.data_mut!();
+    data.version_control().check(VERSION, b"rotate_signers");
+    data.rotate_signers(
+        clock,
         new_signers_data,
-        |bcs| weighted_signers::peel(bcs),
-    );
-    let proof = utils::peel!(proof_data, |bcs| proof::peel(bcs));
-
-    let enforce_rotation_delay = ctx.sender() != self.operator;
-
-    let is_latest_signers = self
-        .signers
-        .validate_proof(
-            data_hash(COMMAND_TYPE_ROTATE_SIGNERS, new_signers_data),
-            proof,
-        );
-    assert!(!enforce_rotation_delay || is_latest_signers, ENotLatestSigners);
-
-    // This will fail if signers are duplicated
-    self
-        .signers
-        .rotate_signers(clock, weighted_signers, enforce_rotation_delay);
+        proof_data,
+        ctx,
+    )
```

Comment:
```suggestion
    let data = self.data_mut!(b"rotate_signers");
    data.rotate_signers(
        clock,
        new_signers_data,
        proof_data,
        ctx,
    )
```

We can perform the version check in the macro itself. Don't even need to create a `data` variable but seems less readable without it.